File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/AddOnPicker.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import AddOn from "@mateu/shared/apiClients/dtos/componentmetadata/AddOn";

export default interface AddOnPicker extends ComponentMetadata {
    totalLabel?: string
    currency?: string
    actionId?: string
    items?: AddOn[]
}

File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/AddOnPicker.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import java.util.List;
import lombok.Builder;

/**
 * Priced extras with a live running total (e.g. upsell packages at check-in): a responsive grid of
 * {@link AddOn} cards, each with a +/✓ toggle, and a header-right total ({@code totalLabel} +
 * {@code currency} + sum of added prices, computed client-side). On every toggle, if {@code
 * actionId} is present it is dispatched with {@code { "_item": id, "_added": bool, "_total": number
 * }}. Design-system neutral, dark-mode aware.
 */
@Builder
public record AddOnPicker(
    String id,
    String totalLabel,
    String currency,
    String actionId,
    List<AddOn> items,
    String style,
    String cssClasses)
    implements Component {}
